From 90e5d05beb010ea967550336036b5156db911c53 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 20 Jun 2024 13:34:48 +0300
Subject: [PATCH] [FSW-2704] fs1: add valid manifest index

In cases where manifest and ppboot validation passes, bootrom
will store valid manifest index in cold reset flops.
This bit value is only valid if validation_pass flag is set.
0 - SPI Manifest 0 is valid
1- SPI Manifest 1 is valid

Change-Id: I242bdf15061d236b3b49cb74e23a295020b3f691
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index c957003..a9d457a 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -14,6 +14,7 @@
 	.config2_fuse = 0,
 	.recovery_flag = 0,
 	.validation_pass = 0,
+	.valid_manifest_idx = 0,
 	.efuse_read = 0,
 	.curr_img = 0,
 	.fake_fuse_en = false,
diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 6912ce6..986e0fb 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -135,6 +135,7 @@
 
 	boot_rom_global.recovery_flag = cold_rst_data.recovery_flag;
 	boot_rom_global.validation_pass = cold_rst_data.validation_flag;
+	boot_rom_global.valid_manifest_idx = cold_rst_data.valid_manifest_idx;
 	boot_rom_global.efuse_read = cold_rst_data.efuse_read_flag;
 	boot_rom_global.boot_info_set_done = cold_rst_data.boot_info_set_done;
 	boot_rom_global.tpm_extend_done = cold_rst_data.tpm_validation_flag;
diff --git a/include/boot_rom.h b/include/boot_rom.h
index 620b740..4a3532d 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -72,6 +72,7 @@
 	uint32_t config2_fuse;
 	uint8_t recovery_flag;
 	uint8_t validation_pass;
+	uint8_t valid_manifest_idx;
 	uint8_t efuse_read;
 	uint8_t curr_img;
 	uint8_t die_id;
